### PR TITLE
feat: allow any layer to be embedding layer

### PR DESCRIPTION
**Pull request by bwanglzu**
- [x] pytorch
- [x] tensorflow
- [x] paddle
- [x] fix all tests

Other changes:
1. `output_dim` is removed, the output dim dependent on the output of user provided `bottleneck_net`.
2. `freeze` semantic change, if `freezze` is bool, will freeze all layers/no freeze, if `freeze` is a list of string, then freeze layers in the list. `freeze_layers` has been removed.

Next steps:
Add `pooling` at the end of the `layer_name`, allow pooling based on `avg`, `max`, `gem` etc. 

example code:
```python
from finetuner.tailor.pytorch import PytorchTailor
from efficientnet_pytorch import EfficientNet
from finetuner import display

import torch.nn as nn

class BottleNeckNet(nn.Module):
    def __init__(self, in_features):
        super().__init__()
        self.flatten = nn.Flatten()
        self.linear1 = nn.Linear(in_features, 2048)
        self.linear2 = nn.Linear(2048, 1024)

    def forward(self, input_):
        return self.linear2(self.linear1(self.flatten(input_)))


model = EfficientNet.from_pretrained('efficientnet-b0')
tailor = PytorchTailor(model, (3, 224, 224))

b = BottleNeckNet(in_features=62720)
tailed_efficientnet = tailor.to_embedding_model(layer_name='conv2dstaticsamepadding_274', bottleneck_net=b, freeze=True)

display(tailed_efficientnet, (3, 224, 224))
```

66DEAA2D-8137-4FFE-9C4C-14F8481C7CD6

### CODE REVIEW

File: finetuner/tailor/base.py [hunk 1]
```diff
@@ -80,11 +80,17 @@ def display(self, *args, **kwargs) -> None:
...
+                style = 'cyan'
+            if 'identity' in s['name']:
+                style = 'bright_black'
             table.add_row(*map(str, (s[v] for v in cols)), style=style)
         print(
             table,
-            '[green]Green[/green] layers can be used as embedding layers, '
-            'whose [b]name[/b] can be used as [b]layer_name[/b] in to_embedding_model(...).',
+            '[green]Green[/green] layers are trainable layers, '
+            '[cyan]Cyan[/cyan] layers are non-trainable layers or freezed layers.\n'
```
Comment (tadejsv) [inline finetuner/tailor/base.py:93]:
```suggestion
            '[cyan]Cyan[/cyan] layers are non-trainable layers or frozen layers.\n'
```

File: finetuner/tailor/base.py [hunk 2]
```diff
@@ -80,11 +80,17 @@ def display(self, *args, **kwargs) -> None:
...
+            '[bright_black]Gray[/bright_black] layers indicates this layer has been replaced as an Identity layer which has no impact on model traning and inference.\n'
```
Comment (tadejsv) [inline finetuner/tailor/base.py:94]:
```suggestion
            '[bright_black]Gray[/bright_black] layers indicates this layer has been replaced by an Identity layer.\n'
```

File: tests/unit/tailor/test_display.py [hunk 3]
```diff
@@ -281,11 +281,12 @@ def test_display_mlp_original(framework, capsys):
     assert _display.strip() in capsys.readouterr().out.strip()
 
 
-@pytest.mark.parametrize('framework', ('paddle', 'pytorch', 'keras'))
+@pytest.mark.parametrize('framework', ('pytorch', 'paddle', 'keras'))
 def test_display_lstm_original(framework, capsys):
     model = get_lstm(framework)
     _display = get_lstm_display(framework)
     display(model, input_size=(100,), input_dtype='int64')
+    print(_display.strip())  # Add intentionally to allow capsys capture the display
```
Comment (tadejsv) [inline tests/unit/tailor/test_display.py:289]: I don't see why this is needed - you already display the model before?

**Review by tadejsv — CHANGES_REQUESTED** [review]

**Review by tadejsv — APPROVED** [review]
LGTM
